### PR TITLE
added --dev flag to the cli

### PR DESCRIPTION
**Pull request by dannyzaken**
### Explain the changes
1. added `--dev` flag to noobaa CLI
2. same as `--mini` but with higher resources requests tuned for a dev environment (mainly M1 macs)

### Issues: Fixed #xxx / Gap #xxx
1. Fixes https://github.com/noobaa/noobaa-operator/issues/1017

### Testing Instructions:
1. 

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

**Review by v-harihar — APPROVED** [review]
just the comment in wrong place. apart from that lgtm

File: pkg/options/options.go [hunk 1]
```diff
@@ -110,6 +110,11 @@ var ImagePullSecret = ""
 // pod)
 var MiniEnv = false
 
+// MiniEnv setting this option indicates to the operator that it is deployed on low resource environment
```
Comment (liranmauda) [inline pkg/options/options.go:113]:
```suggestion
// DevEnv setting this option indicates to the operator that it is deployed on low resource environment
```

**Review by liranmauda — COMMENTED** [review]
@dannyzaken Why should we keep the MiniEnv (--mini) flag?

File: pkg/options/options.go (hunk 1 above)
Comment (shirady) [reply pkg/options/options.go:113]: Please update the text (and not only `MiniEnv` to `DevEnv`).

File: pkg/options/options.go [hunk 2]
```diff
@@ -209,6 +214,10 @@ func init() {
 		&MiniEnv, "mini",
 		false, "Signal the operator that it is running in a low resource environment",
 	)
+	FlagSet.BoolVar(
+		&DevEnv, "dev",
+		false, "Set sufficient resoureces for dev env (Arm Macs)",
```
Comment (shirady) [inline pkg/options/options.go:219]: Please change `Arm` to `ARM`

File: pkg/options/options.go [hunk 3]
```diff
@@ -110,6 +110,11 @@ var ImagePullSecret = ""
 // pod)
 var MiniEnv = false
 
+// MiniEnv setting this option indicates to the operator that it is deployed on low resource environment
+// This info is used by the operator for environment based decisions (e.g. number of resources to request per
+// pod)
+var DevEnv = false
```
Comment (shirady) [inline pkg/options/options.go:116]: Why is it a global flag?
I thought we use it only in `install`...

**Comment by shirady** [comment]
@dannyzaken, would you please add this issue #1017 as fixed in the description?

**Comment by dannyzaken** [comment]
> @dannyzaken Why should we keep the MiniEnv (--mini) flag?

it is used by our tests. at first I tried to increase the resources for `--mini`, but the images in the actions don't have enough resources

File: pkg/options/options.go (hunk 2 above)
Comment (dannyzaken) [reply pkg/options/options.go:219]: decided to drop the reference to arm altogether

**Comment by dannyzaken** [comment]
> General comment - I think [it](https://github.com/noobaa/noobaa-operator/actions/runs/4740342204/jobs/8419275980?pr=1094) is an example of why we should run linters (`make lint` & `golangci-lint `) separated from the tests themselves (we could see that it doesn't break anything and we have just a lint error about a comment).

actually, for the operator we do have a separate action for the linter, but it passed. don't know what are the differences from the makefile one

**Pull request merged**